lib/extractor/oci-archive/layer.ts

```diff
@@ -30,18 +31,20 @@ const MEDIATYPE_OCI_MANIFEST_LIST_V1 =
  * Retrieve the products of files content from the specified oci-archive.
  * @param ociArchiveFilesystemPath Path to image file saved in oci-archive format.
  * @param extractActions Array of pattern-callbacks pairs.
+ * @param platform Platform string.
  * @returns Array of extracted files products sorted by the reverse order of the layers from last to first.
  */
 export async function extractArchive(
   ociArchiveFilesystemPath: string,
   extractActions: ExtractAction[],
+  platform: string,
 ): Promise<ExtractedLayersAndManifest> {
   return new Promise((resolve, reject) => {
     const tarExtractor: Extract = extract();
 
     const layers: Record<string, ExtractedLayers> = {};
     const manifests: Record<string, OciArchiveManifest> = {};
-    let imageConfig: ImageConfig | undefined;
+    const configs: ImageConfig[] = [];
```

Comment:
Multi-platform images have a config for each platform included in the image.
Note: The index file will always list all platforms supported even if not included in the image

Reply:
Sorry, I was not very clear here! When an image has a manifest of type `application/vnd.oci.image.index.v1+json` it has all the available platforms